discordclient/commands/commands.go

```diff
@@ -20,12 +21,38 @@ var commandHandlers = map[string]func(s *dgo.Session, i *dgo.InteractionCreate){
 	repositoryLinkCommand.Name: repositoryLinkHandler,
 	alumniLinkCommand.Name:     alumniLinkHandler,
 	socialNetworkCommand.Name:  socialNetworkHandler,
+	createEventsCommand.Name:   createEventsHandler,
 }
 
 // CommandsHandlers is the handler of slash commands
 func CommandsHandlers(s *dgo.Session, i *dgo.InteractionCreate) {
-	if h, ok := commandHandlers[i.ApplicationCommandData().Name]; ok {
-		h(s, i)
+	switch i.Type {
+	case dgo.InteractionApplicationCommand:
+		if h, ok := commandHandlers[i.ApplicationCommandData().Name]; ok {
+			h(s, i)
+		}
+	case dgo.InteractionModalSubmit:
```

Comment:
There is too much code in this case branch, you handle the interactoin then chech if there is errors, then get the data, then ...
The handler code must juste call a function (that can be tested).

Reply:
yes, it's more a test of how to interact with a model than real production code